examples/camera-controller/main.cpp

```diff
@@ -22,20 +22,45 @@
 #include <commands/interactive/Commands.h>
 #include <commands/pairing/Commands.h>
 #include <commands/webrtc/Commands.h>
+#include <platform/CHIPDeviceLayer.h>
 #include <webrtc_manager/WebRTCManager.h>
 #include <zap-generated/cluster/Commands.h>
 
+#include <csignal>
 #include <iostream>
 #include <string>
+#include <unistd.h>
 #include <vector>
 
 using namespace chip;
 
+namespace {
+
+void StopSignalHandler(int signum)
+{
+    ChipLogProgress(NotSpecified, "SIGTERM received, exiting...");
+
+    DeviceLayer::PlatformMgr().Shutdown();
+    Platform::MemoryShutdown();
```

Comment:
Memory management in a signal handler is not a good idea. The only thing that you should do here is to exit the loop. The reset should be done outside of the signal handler (in the main function after the loop exits).

Reply:
Acked